omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -0,0 +1,1898 @@
+                else:
+                    print("go to approach pose")
+                    yield from self._move_hand_direct_cartesian(approach_pose, ignore_failure=False, stop_if_stuck=True)
+            
+                # Step once to update
+                empty_action = self._empty_action()
+                yield self.with_context(empty_action)
+
+                for i, target_pose in enumerate(target_poses):
+                    print(f"go to target pose {i}")
+                    yield from self._move_hand_direct_cartesian(target_pose, ignore_failure=False, stop_if_stuck=True)
+
+                # Moving to target pose often fails. Let's get the hand to apply the correct actions for its current pos
+                # This prevents the hand from jerking into its desired position when we do a release.
+                yield from self._move_hand_direct_cartesian(
+                    self.robot.eef_links[self.arm].get_position_orientation(), 
+                    ignore_failure=True,
+                    stop_if_stuck=True
+                )
+
+                if should_open:
+                    yield from self._execute_release()
+                    yield from self._move_base_backward()
+
+            except ActionPrimitiveError as e:
+                print(e)
+                if should_open:
+                    yield from self._execute_release() 
+                    yield from self._move_base_backward()
+                else:
+                    yield from self._move_hand_backward()
+
+        if obj.states[object_states.Open].get_value() != should_open:
+            raise ActionPrimitiveError(
+                ActionPrimitiveError.Reason.POST_CONDITION_ERROR,
+                "Despite executing the planned trajectory, the object did not open or close as expected. Maybe try again",
+                {"target object": obj.name, "is it currently open": obj.states[object_states.Open].get_value()},
+            )
+        
+        # Reset object to track
+        self._tracking_object_pose = None
+        self._track_eef = False
+
+
+    def _move_base_backward(self, steps=5, speed=0.2):
+        """
+        Yields action for the robot to move base so the eef is in the target pose using the planner
+
+        Args:
+            steps (int): steps to move base
+            speed (float): base speed
+
+        Returns:
+            np.array or None: Action array for one step for the robot to move base or None if its at the target pose
+        """
+        initial_eef_pos = self.robot.get_eef_position()
+        for _ in range(steps):
+            action = self._empty_action(arm_pose_to_keep=initial_eef_pos)
+            action[self.robot.controller_action_idx["gripper_{}".format(self.arm)]] = 1.0
+            action[self.robot.base_control_idx[0]] = -speed
+            yield self.with_context(action)
+
+    def _move_hand_backward(self, steps=5, speed=0.2):
```

Comment:
maybe generalize to a more general `_move_hand_in_direction(self, axis, steps=5, speed=0.2)` since only one line changes?

Reply:
I think we can change these later. I know it's a bit repetitive but at least it's clear.